src/card_operations/check_pairing.c

```diff
@@ -0,0 +1,150 @@
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "check_pairing.h"
+
+#include "card_internal.h"
+#include "events.h"
+#include "ui_message.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * TODO: add description when this api gets concrete definition
+ *
+ * @param connection_data
+ */
+static void init_tap_card_data(NFC_connection_data *connection_data);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+static void init_tap_card_data(NFC_connection_data *connection_data) {
+  memset(connection_data, 0, sizeof(NFC_connection_data));
+  connection_data->retries = 5;
+  connection_data->desktop_control = false;
+  connection_data->acceptable_cards = ACCEPTABLE_CARDS_ALL;
+  memset(connection_data->family_id, 0xff, sizeof(connection_data->family_id));
+  memset(connection_data->card_key_id, 0, sizeof(connection_data->card_key_id));
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+card_error_type_e card_check_pairing(check_pairing_result_t *result) {
+  LOG_SWV("%s (%d): %p\n", __func__, __LINE__, result);
+  card_error_type_e status = CARD_OPERATION_DEFAULT_INVALID;
+  if (NULL == result) {
+    return status;
+  }
+
+  memset(result, 0, sizeof(check_pairing_result_t));
+  result->card_number = 0;
+  card_operation_data_t operation_data = {{0}, NULL, 0};
+  init_tap_card_data(&operation_data.nfc_data);
+  status = card_initialize_applet(&operation_data);
+  buzzer_start(100);
+  nfc_deselect_card();
+
+  if (CARD_OPERATION_SUCCESS == status) {
```

Comment:
Please unify error handling as done in card authentication by Aman, once that is merged

Reply:
there are no cases that need retrying here. card authentication has some error scenarios where retrying is feasible. such as wrong card sequence, wrong family-id, etch. such cases will not happen in this particular flow.